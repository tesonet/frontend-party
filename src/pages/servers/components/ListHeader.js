import React from 'react'

export const ListHeader = ({ tableColumns, handleSorting }) => (
    <div className="list-header-container">
		    {tableColumns.map((item, index) => (<div key={index} onClick={() => handleSorting(item.sortBy, item.sortingType)}>{item.title}</div>))}
    	</div>
)