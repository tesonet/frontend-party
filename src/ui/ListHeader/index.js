import styled from 'styled-components'
import Box from '../Box'
import theme from '../styles/theme'

const ListHeader = styled(Box)`
  display: flex;
  border-bottom: 1px solid ${theme.palette.grey['300']};
  justify-content: space-between;
  align-items: center;
  background-color: ${theme.palette.grey['100']};
`

export default ListHeader
