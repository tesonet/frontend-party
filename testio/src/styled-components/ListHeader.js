import styled from 'styled-components'
export const ListHeader = styled.li`
    height: 57.99px;
    border-bottom:1px solid #e5e5e5;
    background-color:#f5f5f5;
    color:#999999;
    text-transform: uppercase;
    
`;
