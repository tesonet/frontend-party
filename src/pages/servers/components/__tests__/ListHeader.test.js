import { ListHeader } from '../ListHeader';

const tableColumns = [{
        title: 'Name',
        sortBy: 'name',
        sortingType: 'alphabetically'
    },
    {
        title: 'Distance',
        sortBy: 'distance',
        sortingType: 'numbers'
    }
];

it("should match snapshot", () => {
    const wrapper = shallow(
        <ListHeader tableColumns={tableColumns} handleSorting={() => {}}/>
    );
    expect(wrapper).toMatchSnapshot();
});

it("should add list headers", () => {
    const wrapper = mount(
        <ListHeader tableColumns={tableColumns} handleSorting={() => {}}/>
    );

    expect(wrapper.find('.list-header-container').children()).toHaveLength(tableColumns.length);
});
