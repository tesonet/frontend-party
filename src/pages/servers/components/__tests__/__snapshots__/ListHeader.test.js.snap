// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="list-header-container"
>
  <div
    key="0"
    onClick={[Function]}
  >
    Name
  </div>
  <div
    key="1"
    onClick={[Function]}
  >
    Distance
  </div>
</div>
`;
